packages/grafana-runtime/src/services/dataSourceSrv.ts

```diff
@@ -19,6 +19,16 @@ export interface DataSourceSrv {
    * Returns metadata based on UID.
    */
   getDataSourceSettingsByUid(uid: string): DataSourceInstanceSettings | undefined;
+
+  /**
+   * Get all data sources
+   */
+  getAll(): DataSourceInstanceSettings[];
+
+  /**
+   * Get all data sources except for internal ones that usually should not be listed like mixed data source.
+   */
+  getExternal(): DataSourceInstanceSettings[];
```

Comment:
@dprokop @torkelo does it make sense to add it here? Otherwise I would have to use the grafana main package instead of runtime and these seems like reasonable to have here like creating a 3rd party "mixed" datasource.

Reply:
yes, makes sense :+1: